raiden/transfer/channel.py

```diff
@@ -1213,39 +1183,35 @@ def update_contract_balance(end_state: NettingChannelEndState, contract_balance:
         end_state.contract_balance = contract_balance
 
 
-def compute_merkletree_with(
-    merkletree: MerkleTreeState, lockhash: LockHash
-) -> Optional[MerkleTreeState]:
-    """Register the given lockhash with the existing merkle tree."""
-    # Use None to inform the caller the lockshash is already known
-    result = None
-
-    leaves = merkletree.layers[LEAVES]
-    if lockhash not in leaves:
-        leaves = list(leaves)
-        leaves.append(Keccak256(lockhash))
-        result = MerkleTreeState(compute_layers(leaves))
-
-    return result
-
+def compute_locks_with(
+    locks: PendingLocksState, lock: Union[HashTimeLockState, UnlockPartialProofState]
+) -> Optional[PendingLocksState]:
+    """Register the given lock with as a pending locks."""
+    if bytes(lock.encoded) not in locks.locks:
+        locks = PendingLocksState(list(locks.locks))
```

Comment:
Why construct a new state if you already have `locks`?

Reply:
Exactly the same reason as above https://github.com/raiden-network/raiden/pull/4192#discussion_r294872246 .